### PR TITLE
[otp_ctrl] Update checklist and move to D3

### LINKED ISSUE #13438: [otp_ctrl] D3 opens
- [x] FSM coverage should work now with the sparse FSM workaround. @cindychip  to confirm on https://github.com/lowRISC/opentitan/issues/11607.
- [x] Limited updates to the memory map (e.g. changing partition size, or modifying HW_CFG / SW_CFG items) should not gate D3 (this would be similar to parameterizing a PLIC or alert handler for a specific top-level. However, we should add a documentation step to the D3 process where we explicitly list out which post D3 re-parameterizations are ok. -> spun out into a separate issue: #13858
- [x] spell out which post D3 re-parameterizations are fine and don’t require a bounce back to D2S #13860.
- [x] Add a closed source test point for testing the Vendor test GPIO @cindychip 
- [x] Fix all lint errors
- [x] Try to use IPgen to produce OTP_CTRL -> since this does not change the RTL it is ok to split this out into a separate task that should be done by M3: https://github.com/lowRISC/opentitan/issues/8440
- [x] Resolve https://github.com/lowRISC/opentitan/pull/13043 
- [x] Audit the sram adapter configuration used within OTP #9438
- [x] fix bus integrity error according to https://github.com/lowRISC/opentitan/issues/13366 https://github.com/lowRISC/opentitan/issues/14472 #14720

### PR DESCRIPTION
**Pull request by msfschaffner**
To be merged once all cleanup items have been addressed.

Fix #13438

Signed-off-by: Michael Schaffner <msf@opentitan.org>

### CODE REVIEW

**Comment by msfschaffner** [comment]
This should be ready to merge once #14719 and #14720 have landed.

**Comment by tjaychen** [comment]
yay! congrats!

**Review by tjaychen — APPROVED** [review]

**Review by GregAC — APPROVED** [review]

**Review by a-will — APPROVED** [review]

**Pull request merged**
